File: src/lib/categories.ts
import type { CategoryMeta, ProductCategory } from "@/types";

/** Canonical Hubsom categories — used across Buy Now, live, auctions, flash, stores. */
export const CATEGORIES: CategoryMeta[] = [
  { slug: "groceries", name: "Groceries", description: "Food and everyday essentials — Buy Now, live, or auctioned." },
  { slug: "electronics", name: "Electronics", description: "Gadgets, audio, and consumer electronics." },
  { slug: "fashion", name: "Fashion", description: "Apparel and style from Accra to Kumasi." },
  { slug: "shoes", name: "Shoes", description: "Sneakers, sandals, and formal footwear." },
  { slug: "beauty-personal-care", name: "Beauty & Personal Care", description: "Skincare, makeup, and grooming." },
  { slug: "health-wellness", name: "Health & Wellness", description: "Wellness products and personal health." },
  { slug: "home-kitchen", name: "Home & Kitchen", description: "Cookware, tableware, and household essentials." },
  { slug: "furniture", name: "Furniture", description: "Home and office furniture." },
  { slug: "home-decor", name: "Home Décor", description: "Artful accents for living spaces." },
  { slug: "appliances", name: "Appliances", description: "Kitchen and home appliances." },
  { slug: "phones-accessories", name: "Phones & Accessories", description: "Smartphones, cases, and mobile gear." },
  { slug: "computers-tablets", name: "Computers & Tablets", description: "Laptops, tablets, and peripherals." },
  { slug: "gaming", name: "Gaming", description: "Consoles, games, and gaming gear." },
  { slug: "cameras-photography", name: "Cameras & Photography", description: "Cameras, lenses, and photo gear." },
  { slug: "jewelry-watches", name: "Jewelry & Watches", description: "Jewelry, beads, and timepieces." },
  { slug: "luxury-goods", name: "Luxury Goods", description: "Premium bags, watches, and exclusives." },
  { slug: "baby-kids", name: "Baby & Kids", description: "Care, gear, and essentials for little ones." },
  { slug: "toys-games", name: "Toys & Games", description: "Toys, puzzles, and play sets." },
  { slug: "sports-outdoors", name: "Sports & Outdoors", description: "Fitness gear and outdoor kit." },
  { slug: "automotive", name: "Automotive", description: "Parts, accessories, and vehicle care." },
  { slug: "tools-hardware", name: "Tools & Hardware", description: "Hand tools, power tools, and hardware." },
  { slug: "pet-supplies", name: "Pet Supplies", description: "Food, toys, and care for pets." },
  { slug: "books", name: "Books", description: "Education, fiction, and local titles." },
  { slug: "music-instruments", name: "Music & Instruments", description: "Instruments, audio, and sheet music." },
  { slug: "movies-entertainment", name: "Movies & Entertainment", description: "Films, media, and entertainment." },
  { slug: "art-collectibles", name: "Art & Collectibles", description: "Original works, prints, and rare finds." },
  { slug: "antiques-vintage", name: "Antiques & Vintage", description: "Timeless second-hand treasures." },
  { slug: "handmade-crafts", name: "Handmade & Crafts", description: "Crafted goods from local makers." },
  { slug: "office-school-supplies", name: "Office & School Supplies", description: "Stationery and workspace tools." },
  { slug: "garden-outdoor", name: "Garden & Outdoor", description: "Plants, outdoor living, and garden tools." },
  { slug: "industrial-business-equipment", name: "Industrial & Business Equipment", description: "Tools and equipment for trade." },
  { slug: "digital-products", name: "Digital Products", description: "Downloads, courses, and licenses." },
  { slug: "services", name: "Services", description: "Bookable services from trusted providers." },
  { slug: "real-estate", name: "Real Estate", description: "Listings and property opportunities." },
  { slug: "vehicles", name: "Vehicles", description: "Cars, motorbikes, and mobility." },
  { slug: "tickets-events", name: "Tickets & Events", description: "Event tickets and experiences." },
  { slug: "gift-cards", name: "Gift Cards", description: "Digital and physical gift cards." },
  { slug: "miscellaneous", name: "Miscellaneous", description: "Everything else that belongs on Hubsom." },
];

export const CATEGORY_MAP = Object.fromEntries(
  CATEGORIES.map((c) => [c.slug, c]),
) as Record<ProductCategory, CategoryMeta>;

export const ALL_FULFILLMENT_MODES = [
  "buy-now",
  "live-selling",
  "live-auction",
  "flash-sale",
  "bundle",
  "store-listing",
  "promotion",
] as const;

export function categoryName(slug: ProductCategory | string): string {
  return CATEGORY_MAP[slug as ProductCategory]?.name ?? slug;
}

File: hubsom_app/lib/core/constants/categories.dart
import 'package:flutter/material.dart';

class CategoryMeta {
  const CategoryMeta(this.slug, this.name, this.description, this.icon);
  final String slug;
  final String name;
  final String description;
  final IconData icon;
}

const hubsomCategories = <CategoryMeta>[
  CategoryMeta('groceries', 'Groceries', 'Fresh food and pantry staples', Icons.local_grocery_store_outlined),
  CategoryMeta('electronics', 'Electronics', 'Gadgets and devices', Icons.devices_outlined),
  CategoryMeta('fashion', 'Fashion', 'Clothing and apparel', Icons.checkroom_outlined),
  CategoryMeta('shoes', 'Shoes', 'Footwear for every style', Icons.directions_walk_outlined),
  CategoryMeta('beauty-personal-care', 'Beauty & Personal Care', 'Skincare and grooming', Icons.spa_outlined),
  CategoryMeta('health-wellness', 'Health & Wellness', 'Supplements and care', Icons.favorite_outline),
  CategoryMeta('home-kitchen', 'Home & Kitchen', 'Kitchen and home essentials', Icons.kitchen_outlined),
  CategoryMeta('furniture', 'Furniture', 'Home and office furniture', Icons.chair_outlined),
  CategoryMeta('home-decor', 'Home Decor', 'Decor and accents', Icons.light_outlined),
  CategoryMeta('appliances', 'Appliances', 'Home appliances', Icons.microwave_outlined),
  CategoryMeta('phones-accessories', 'Phones & Accessories', 'Phones and accessories', Icons.smartphone_outlined),
  CategoryMeta('computers-tablets', 'Computers & Tablets', 'Laptops and tablets', Icons.laptop_outlined),
  CategoryMeta('gaming', 'Gaming', 'Games and consoles', Icons.sports_esports_outlined),
  CategoryMeta('cameras-photography', 'Cameras & Photography', 'Cameras and gear', Icons.photo_camera_outlined),
  CategoryMeta('jewelry-watches', 'Jewelry & Watches', 'Jewelry and timepieces', Icons.watch_outlined),
  CategoryMeta('luxury-goods', 'Luxury Goods', 'Premium brands', Icons.diamond_outlined),
  CategoryMeta('baby-kids', 'Baby & Kids', 'Baby and kids essentials', Icons.child_friendly_outlined),
  CategoryMeta('toys-games', 'Toys & Games', 'Toys and games', Icons.toys_outlined),
  CategoryMeta('sports-outdoors', 'Sports & Outdoors', 'Sporting goods', Icons.sports_soccer_outlined),
  CategoryMeta('automotive', 'Automotive', 'Car parts and accessories', Icons.build_outlined),
  CategoryMeta('tools-hardware', 'Tools & Hardware', 'Tools and hardware', Icons.handyman_outlined),
  CategoryMeta('pet-supplies', 'Pet Supplies', 'Pet food and accessories', Icons.pets_outlined),
  CategoryMeta('books', 'Books', 'Books and reading', Icons.menu_book_outlined),
  CategoryMeta('music-instruments', 'Music & Instruments', 'Instruments and gear', Icons.music_note_outlined),
  CategoryMeta('movies-entertainment', 'Movies & Entertainment', 'Movies and media', Icons.movie_outlined),
  CategoryMeta('art-collectibles', 'Art & Collectibles', 'Art and collectibles', Icons.palette_outlined),
  CategoryMeta('antiques-vintage', 'Antiques & Vintage', 'Vintage finds', Icons.museum_outlined),
  CategoryMeta('handmade-crafts', 'Handmade Crafts', 'Handmade goods', Icons.volunteer_activism_outlined),
  CategoryMeta('office-school-supplies', 'Office & School', 'Office and school supplies', Icons.school_outlined),
  CategoryMeta('garden-outdoor', 'Garden & Outdoor', 'Garden and outdoor living', Icons.yard_outlined),
  CategoryMeta('industrial-business-equipment', 'Industrial & Business', 'Business equipment', Icons.precision_manufacturing_outlined),
  CategoryMeta('digital-products', 'Digital Products', 'Digital downloads', Icons.cloud_download_outlined),
  CategoryMeta('services', 'Services', 'Local services', Icons.miscellaneous_services_outlined),
  CategoryMeta('real-estate', 'Real Estate', 'Property listings', Icons.home_work_outlined),
  CategoryMeta('vehicles', 'Vehicles', 'Cars and vehicles', Icons.directions_car_outlined),
  CategoryMeta('tickets-events', 'Tickets & Events', 'Event tickets', Icons.confirmation_number_outlined),
  CategoryMeta('gift-cards', 'Gift Cards', 'Gift cards', Icons.card_giftcard_outlined),
  CategoryMeta('miscellaneous', 'Miscellaneous', 'Everything else', Icons.category_outlined),
];
